diag_manager/fms_diag_file_object.F90

```diff
@@ -1816,6 +1829,13 @@ subroutine close_diag_file(this, output_buffers, model_end_time, diag_fields)
     call close_file(fms2io_fileobj)
   end select
 
+  !< Keep track of the number of time levels that were written to the file
+  !! If the file is using the new_file_freq key, it will be closing the file multiple
```

Comment:
If someone is using the `new_file_freq_key` how is the filename for the `fms_yaml.out` going to increment to keep up with all of the different timestamped versions?  Will there be one yaml specification file with all incrementals or one for each incremental file as it is closed?

Reply:
It is going to increment to keep up with all of the different timestamped version. 

So for example, for this diag_file
```
- file_name: file9%4yr%2mo%2dy%2hr%2min
  filename_time: begin
  freq: 1 hours,1 hours,1 hours
  time_units: hours
  unlimdim: time
  new_file_freq: 6 hours, 3 hours, 1 hours
  start_time: 2 1 1 0 0 0
  file_duration: 12 hours, 3 hours, 9 hours
```

and if the "simulation" is for a day, `number_of_timelevels` is going to be set to 24